### PR TITLE
WorldBorder Support

### PR DESCRIPTION
**Pull request by ItchyDitchy**
## Description

Androids can go past and do stuff outside the world border.

## Proposed changes

Added a checker if block location is outside world border, if true, return (or do not continue)

## Related Issues (if applicable)

None

## Checklist

- [ / ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ / ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [ / ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ / ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/FarmerAndroid.java [hunk 1]
```diff
@@ -37,6 +37,9 @@ protected void farm(Block b, BlockMenu menu, Block block, boolean isAdvanced) {
         BlockData data = block.getBlockData();
         ItemStack drop = null;
 
+        if (!block.getWorld().getWorldBorder().isInside(block.getLocation()))
+        	return;
```
Comment (martinbrom) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/FarmerAndroid.java:41]:
```suggestion
        if (!block.getWorld().getWorldBorder().isInside(block.getLocation())) {
            return;
        }
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/MinerAndroid.java [hunk 2]
```diff
@@ -74,6 +74,9 @@ public AndroidType getAndroidType() {
     protected void dig(Block b, BlockMenu menu, Block block) {
         Collection<ItemStack> drops = block.getDrops(effectivePickaxe);
 
+        if (!block.getWorld().getWorldBorder().isInside(block.getLocation()))
+        	return;
```
Comment (martinbrom) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/MinerAndroid.java:78]:
```suggestion
        if (!block.getWorld().getWorldBorder().isInside(block.getLocation())) {
            return;
        }
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/MinerAndroid.java [hunk 3]
```diff
@@ -99,6 +102,9 @@ protected void dig(Block b, BlockMenu menu, Block block) {
     protected void moveAndDig(Block b, BlockMenu menu, BlockFace face, Block block) {
         Collection<ItemStack> drops = block.getDrops(effectivePickaxe);
 
+        if (!block.getWorld().getWorldBorder().isInside(block.getLocation()))
+        	return;
```
Comment (martinbrom) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/MinerAndroid.java:106]:
```suggestion
        if (!block.getWorld().getWorldBorder().isInside(block.getLocation())) {
            return;
        }
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/WoodcutterAndroid.java [hunk 4]
```diff
@@ -43,7 +43,9 @@ public AndroidType getAndroidType() {
     @Override
     protected boolean chopTree(Block b, BlockMenu menu, BlockFace face) {
         Block target = b.getRelative(face);
-
+        if (!target.getWorld().getWorldBorder().isInside(target.getLocation()))
+        	return true;
```
Comment (martinbrom) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/WoodcutterAndroid.java:47]:
```suggestion
        if (!target.getWorld().getWorldBorder().isInside(target.getLocation())) {
            return true;
        }
```

**Review by martinbrom — CHANGES_REQUESTED** [review]
Code style:

spaces, not tabs
conditions and loops always with braces

**Comment by martinbrom** [comment]
For next time, there's a button to push all suggestions at once.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/MinerAndroid.java [hunk 5]
```diff
@@ -98,7 +97,7 @@ protected void dig(Block b, BlockMenu menu, Block block) {
     @ParametersAreNonnullByDefault
     protected void moveAndDig(Block b, BlockMenu menu, BlockFace face, Block block) {
         Collection<ItemStack> drops = block.getDrops(effectivePickaxe);
-
+
```
Comment (svr333) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/MinerAndroid.java:100]:
```suggestion

```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java [hunk 6]
```diff
@@ -888,6 +888,11 @@ public void addItems(Block b, ItemStack... items) {
     @ParametersAreNonnullByDefault
     protected void move(Block b, BlockFace face, Block block) {
         if (block.getY() > 0 && block.getY() < block.getWorld().getMaxHeight() && block.isEmpty()) {
+            
+            if (!block.getWorld().getWorldBorder().isInside(block.getLocation())) {
+        	    return;
```
Comment (svr333) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java:893]: tab -> spaces
```suggestion
                return;
```

**Review by svr333 — CHANGES_REQUESTED** [review]

**Review by ItchyDitchy — COMMENTED** [review]
Commited

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/MinerAndroid.java [hunk 7]
```diff
@@ -73,7 +73,7 @@ public AndroidType getAndroidType() {
     @ParametersAreNonnullByDefault
     protected void dig(Block b, BlockMenu menu, Block block) {
         Collection<ItemStack> drops = block.getDrops(effectivePickaxe);
-
+        
         if (!SlimefunTag.UNBREAKABLE_MATERIALS.isTagged(block.getType()) && !drops.isEmpty()) {
             OfflinePlayer owner = Bukkit.getOfflinePlayer(UUID.fromString(BlockStorage.getLocationInfo(b.getLocation(), "owner")));
```
Comment (svr333) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/MinerAndroid.java:78]: Remove trailing spaces
```suggestion

        if (!SlimefunTag.UNBREAKABLE_MATERIALS.isTagged(block.getType()) && !drops.isEmpty()) {
            OfflinePlayer owner = Bukkit.getOfflinePlayer(UUID.fromString(BlockStorage.getLocationInfo(b.getLocation(), "owner")));
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/MinerAndroid.java [hunk 8]
```diff
@@ -111,8 +109,7 @@ protected void moveAndDig(Block b, BlockMenu menu, BlockFace face, Block block)
                 }
 
                 // We only want to break non-Slimefun blocks
-                if (!BlockStorage.hasBlockInfo(block)) {
-                    block.getWorld().playEffect(block.getLocation(), Effect.STEP_SOUND, block.getType());
+                if (!BlockStorage.hasBlockInfo(block)) {
```
Comment (svr333) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/MinerAndroid.java:112]: Remove trailing spaces
```suggestion
                if (!BlockStorage.hasBlockInfo(block)) {
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/MinerAndroid.java [hunk 9]
```diff
@@ -98,7 +97,6 @@ protected void dig(Block b, BlockMenu menu, Block block) {
     @ParametersAreNonnullByDefault
     protected void moveAndDig(Block b, BlockMenu menu, BlockFace face, Block block) {
         Collection<ItemStack> drops = block.getDrops(effectivePickaxe);
-
         if (!SlimefunTag.UNBREAKABLE_MATERIALS.isTagged(block.getType()) && !drops.isEmpty()) {
             OfflinePlayer owner = Bukkit.getOfflinePlayer(UUID.fromString(BlockStorage.getLocationInfo(b.getLocation(), "owner")));
```
Comment (svr333) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/MinerAndroid.java:101]: Add removed line back
```suggestion

        if (!SlimefunTag.UNBREAKABLE_MATERIALS.isTagged(block.getType()) && !drops.isEmpty()) {
            OfflinePlayer owner = Bukkit.getOfflinePlayer(UUID.fromString(BlockStorage.getLocationInfo(b.getLocation(), "owner")));
```

**Review by svr333 — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/WoodcutterAndroid.java [hunk 10]
```diff
@@ -43,7 +43,10 @@ public AndroidType getAndroidType() {
     @Override
     protected boolean chopTree(Block b, BlockMenu menu, BlockFace face) {
         Block target = b.getRelative(face);
-
+        if (!target.getWorld().getWorldBorder().isInside(target.getLocation())) {
+            return true;
```
Comment (svr333) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/WoodcutterAndroid.java:47]:
```suggestion

        if (!target.getWorld().getWorldBorder().isInside(target.getLocation())) {
            return true;
```

**Review by svr333 — CHANGES_REQUESTED** [review]

**Review by svr333 — APPROVED** [review]

**Review by ItchyDitchy — COMMENTED** [review]
Implemented

**Review by WalshyDev — APPROVED** [review]

**Pull request merged**